addons/binding/org.openhab.binding.sensebox/src/main/java/org/openhab/binding/sensebox/handler/SenseBoxHandler.java

```diff
@@ -0,0 +1,374 @@
+            }
+        } catch (IOException e) {
+            logger.debug("IO problems while fetching data: {} / {}", query, e.getMessage());
+        }
+
+        return result;
+    }
+
+    private void publishDataForChannel(ChannelUID channelUID) {
+        switch (channelUID.getId()) {
+            case CHANNEL_NAME:
+                updateState(channelUID, stringFromData(data.getName()));
+                break;
+            case CHANNEL_LOCATION:
+                updateState(channelUID, locationFromData(data.getLocation()));
+                break;
+            case CHANNEL_EXPOSURE:
+                updateState(channelUID, stringFromData(data.getExposure()));
+                break;
+            case CHANNEL_API_URL:
+                updateState(channelUID, stringFromData(data.getDescriptor().getApiUrl()));
+                break;
+            case CHANNEL_IMAGE_URL:
+                updateState(channelUID, stringFromData(data.getDescriptor().getImageUrl()));
+                break;
+            case CHANNEL_MAP_URL:
+                updateState(channelUID, stringFromData(data.getDescriptor().getMapUrl()));
+                break;
+            case CHANNEL_UV_INTENSITY:
+                updateState(channelUID, decimalFromSensor(data.getUvIntensity()));
+                break;
+            case CHANNEL_LUMINANCE:
+                updateState(channelUID, decimalFromSensor(data.getLuminance()));
+                break;
+            case CHANNEL_PRESSURE:
+                updateState(channelUID, decimalFromSensor(data.getPressure()));
+                break;
+            case CHANNEL_HUMIDITY:
+                updateState(channelUID, decimalFromSensor(data.getHumidity()));
+                break;
+            case CHANNEL_TEMPERATURE:
+                updateState(channelUID, decimalFromSensor(data.getTemperature()));
+                break;
+            case CHANNEL_UV_INTENSITY_LR:
+                updateState(channelUID, dateTimeFromSensor(data.getUvIntensity()));
+                break;
+            case CHANNEL_LUMINANCE_LR:
+                updateState(channelUID, dateTimeFromSensor(data.getLuminance()));
+                break;
+            case CHANNEL_PRESSURE_LR:
+                updateState(channelUID, dateTimeFromSensor(data.getPressure()));
+                break;
+            case CHANNEL_HUMIDITY_LR:
+                updateState(channelUID, dateTimeFromSensor(data.getHumidity()));
+                break;
+            case CHANNEL_TEMPERATURE_LR:
+                updateState(channelUID, dateTimeFromSensor(data.getTemperature()));
+                break;
+            default:
+                logger.debug("Command received for an unknown channel: {}", channelUID.getId());
+                break;
+        }
+    }
+
+    private State dateTimeFromSensor(SenseBoxSensor data) {
+        State result = UnDefType.UNDEF;
+
+        if (data != null) {
+            if (data.getLastMeasurement() != null) {
```

Comment:
Also a question of coding style, I find multiple nested ifs help readibility, and actual debugging as every clause is on its own line.